Packs/IronscalesEventCollector/Integrations/IronscalesEventCollector/IronscalesEventCollector.py

```diff
@@ -0,0 +1,272 @@
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+import copy
+import dateparser
+import urllib3
+from typing import Dict, Tuple
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+""" CONSTANTS """
+
+VENDOR = "ironscales"
+PRODUCT = "ironscales"
+DEFAULT_FIRST_FETCH = "3 days"
+DEFAULT_MAX_FETCH = 1000
+DEFAULT_LIMIT = 10
+
+
+""" CLIENT CLASS """
+
+
+class Client(BaseClient):  # pragma: no cover
+    def __init__(
+        self,
+        company_id: str,
+        base_url: str,
+        verify_certificate: bool,
+        proxy: bool,
+        api_key: str,
+        scopes: List[str],
+    ) -> None:
+        self.company_id = company_id
+        super().__init__(base_url, verify_certificate, proxy)
+        self._headers = {"Authorization": f'JWT {self.get_jwt_token(api_key, scopes)}'}
+
+    def client_error_handler(self, res) -> Any:
+        try:
+            err_msg = f"Error in API call [{res.status_code}] - {res.json().get('message')}"
+            raise DemistoException(err_msg, res=res)
+        except ValueError:
+            super().client_error_handler(res)
+
+    def get_jwt_token(self, api_key: str, scopes: list) -> Dict[str, Any]:
+        try:
+            jwt_key = self._http_request(
+                method="POST",
+                url_suffix="/get-token/",
+                json_data={"key": api_key, "scopes": scopes},
+            )
+            return jwt_key["jwt"]
+        except DemistoException as e:
+            if "No company found for API key" in str(e):
+                raise DemistoException(
+                    "Authorization Error: make sure the API is set correctly"
+                )
+            raise e
+
+    def get_incident(self, incident_id: int) -> Dict[str, Any]:
+        return self._http_request(
+            method="GET",
+            url_suffix=f"/incident/{self.company_id}/details/{incident_id}",
+        )
+
+    def get_open_incident_ids(self) -> List[int]:
+        return self._http_request(
+            method="GET",
+            url_suffix=f"/incident/{self.company_id}/open/",
+        ).get("incident_ids") or []
+
+
+""" HELPER FUNCTIONS """
+
+
+def get_incident_ids_by_time(
```

Comment:
You say this is called only in the first fetch, is it? can you explain why would it be called only then? From what I am seeing it will always call this function.

Reply:
:itsabug: